experimental/BasisLieHighestWeight/src/ModuleData.jl

```diff
@@ -0,0 +1,86 @@
+abstract type ModuleData end
+
+# To be implemented by subtypes:
+# Mandatory:
+#   dim(M::MyModuleData) -> ZZRingElem
+#   character(M::MyModuleData) -> Dict{WeightLatticeElem,<:IntegerUnion}
+
+
+mutable struct SimpleModuleData <: ModuleData
+    L::LieAlgebra
+    highest_weight::WeightLatticeElem
+
+    # The following fields are not set by default just set for caching
+    dim::ZZRingElem
+    character::Dict{WeightLatticeElem,<:IntegerUnion}
+
+    function SimpleModuleData(L::LieAlgebra, highest_weight::WeightLatticeElem)
+        new(L, highest_weight)
+    end
+end
+
+function L(M::SimpleModuleData)
```

Comment:
for the type `LieAlgebraModule` of "real modules" (concretely given via structure constants, not just some data like here), the corresponding function is called `base_lie_algebra`. For consistency, this name should be used here as well